contrib/windows/exe_path_wrapper.c

```diff
@@ -0,0 +1,81 @@
+#include <windows.h>
+#include <tchar.h>
+#include <stdio.h>
+#include <shlwapi.h>
+
+/* The maximum path length we'll allow */
+#define MAX_PATH_LEN 1920
+
+/* PATH_ENTRIES is our simulated RPATH, usually of the form 'TEXT("../path1"), TEXT("../path2"), ...' */
+#ifndef PATH_ENTRIES
+#define PATH_ENTRIES  TEXT("")
+#endif
+
+LPWSTR pathEntries[] = {
+    PATH_ENTRIES
+};
+
+/* JULIA_EXE_PATH is the relative path to julia.exe */
+#ifndef JULIA_EXE_PATH
+#define JULIA_EXE_PATH "../libexec/julia.exe"
+#endif
+
+int wmain(int argc, wchar_t *argv[], wchar_t *envp[]) {
+    // Determine absolute path to true julia.exe sitting in `libexec/`
+    WCHAR currFileDir[MAX_PATH_LEN];
+    WCHAR juliaPath[MAX_PATH_LEN];
+    if (!GetModuleFileName(NULL, currFileDir, MAX_PATH_LEN)) {
+        fprintf(stderr, "ERROR: GetModuleFileName() failed with code %d\n", GetLastError());
+        exit(1);
+    }
+    PathRemoveFileSpec(currFileDir);
+    PathCombine(juliaPath, currFileDir, TEXT(JULIA_EXE_PATH));
+
+    // On windows, we simulate RPATH by pushing onto PATH
+    LPWSTR pathVal = (LPWSTR) malloc(MAX_PATH_LEN*sizeof(WCHAR));
+    DWORD dwRet = GetEnvironmentVariable(TEXT("PATH"), pathVal, MAX_PATH_LEN);
+    DWORD numPathEntries = sizeof(pathEntries)/sizeof(pathEntries[0]);
+    if (dwRet == 0) {
+        // If we cannot get PATH, then our job is easy!
+        pathVal[0] = '\0';
+    }
+    else {
+        // Otherwise, we append, if we have enough space to:
+        DWORD currFileDirLen = wcslen(currFileDir);
+        DWORD totalPathLen = dwRet + 1 + currFileDirLen;
+        int env_idx;
+        for (env_idx = 0; env_idx < numPathEntries; ++env_idx) {
+            totalPathLen += 1 + currFileDirLen + 1 + wcslen(pathEntries[env_idx]);
+        }
+        if (MAX_PATH_LEN < totalPathLen) {
+            fprintf(stderr, "ERROR: Cannot append entries to PATH: not enough space in environment block.  Reduce size of PATH!");
```

Comment:
I'm just concerned that it may not be unusual for some user paths to be already near this limit due to the system configuration (that the user might not be able to alter)

Reply:
I don't have any stats on this so you may be right, but I don't know what we can do about it; I haven't been able to come up with any valid alternatives.